src/main/java/lv/ctco/javaschool/goal/boundary/GoalApi.java

```diff
@@ -1,31 +1,146 @@
 
 @Path("/goal")
 @Stateless
 public class GoalApi {
-    @PersistenceContext
-    private EntityManager em;
     @Inject
     private UserStore userStore;
     @Inject
     private GoalStore goalStore;
 
+    @GET
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/mygoals")
+    public List<GoalDto> getMyGoals() {
+        User currentUser = userStore.getCurrentUser();
+        List<Goal> goalsList = goalStore.getGoalsListFor(currentUser);
+        if (goalsList.size() != 0) {
+            return goalsList.stream()
+                    .map(this::convertToDto)
+                    .collect(Collectors.toList());
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    @GET
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/mygoals/{id}")
+    public GoalDto getGoalDtoByGoalId(@PathParam("id") long goalId) {
+        Optional<Goal> goal = goalStore.getGoalById(goalId);
+        if (goal.isPresent()) {
+            Goal g = goal.get();
+            return convertToDto(g);
+        } else {
+            return new GoalDto();
+        }
+    }
+
     @POST
     @RolesAllowed({"ADMIN", "USER"})
-    public void startPage() {
-        /// place for methods on page reload
-        /// currently - none
+    @Path("/newgoal")
+    public void createNewGoal(GoalFormDto goalDto) {
+        User user = userStore.getCurrentUser();
+        Goal goal = new Goal();
+        if (!goalDto.getGoalMessage().isEmpty() && !goalDto.getDeadline().isEmpty()) {
+            goal.setGoalMessage(goalDto.getGoalMessage());
+            goal.setTags(parseStringToTags(goalDto.getGoalMessage()));
+
+            DateTimeFormatter formatterD = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+            LocalDate localDate = LocalDate.parse(goalDto.getDeadline(), formatterD);
+            goal.setDeadlineDate(localDate);
+
+            goal.setUser(user);
+            goal.setRegisteredDate(LocalDateTime.now());
+            goalStore.addGoal(goal);
+        }
+    }
+
+    GoalDto convertToDto(Goal goal) {
+        GoalDto dto = new GoalDto();
+        dto.setUsername(goal.getUser().getUsername());
+        dto.setGoalMessage(goal.getGoalMessage());
+        dto.setDeadlineDate(convertDate(goal.getDeadlineDate()));
+        dto.setRegisteredDate(convertDateTime(goal.getRegisteredDate()));
+        dto.setDaysLeft(countDaysLeft(goal.getDeadlineDate()));
+        dto.setId(goal.getId());
+        return dto;
+    }
+
+    long countDaysLeft(LocalDate deadlineDate) {
+        LocalDate localDate = LocalDate.now();
+        return DAYS.between(localDate, deadlineDate);
+    }
+
+    String convertDate(LocalDate date) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        return date.format(formatter);
+    }
+
+    String convertDateTime(LocalDateTime date) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy kk:mm");
+        return date.format(formatter);
+    }
```

Comment:
These three functions are independent and look like they belong in an utility class that works with Dates.